src/transport/SessionHolder.h

```diff
@@ -51,10 +51,10 @@ class SessionHolder : public SessionDelegate, public IntrusiveListNodeBase
     void Release();
 
     operator bool() const { return mSession.HasValue(); }
-    SessionHandle Get() const { return SessionHandle{ mSession.Value().Get() }; }
-    Optional<SessionHandle> ToOptional() const
+    Optional<SessionHandle> Get() const
     {
-        return mSession.HasValue() ? chip::MakeOptional<SessionHandle>(Get()) : chip::Optional<SessionHandle>::Missing();
+        return mSession.HasValue() ? chip::MakeOptional<SessionHandle>(mSession.Value().Get())
+                                   : chip::Optional<SessionHandle>::Missing();
```

Comment:
Why can't this all just be:
```suggestion
        return mSession;
```
?  Worth at least a comment.